pkg/services/ngalert/api/api_testing.go

```diff
@@ -77,3 +78,18 @@ func (srv TestingApiSrv) RouteTestRuleConfig(c *models.ReqContext, body apimodel
 		nil,
 	)
 }
+
+func (srv TestingApiSrv) RouteEvalQueries(c *models.ReqContext, cmd apimodels.EvalQueriesPayload) response.Response {
```

Comment:
Is there proper permission checks in place to make sure someone cannot query a data source that they don't have permission to?

Reply:
Good catch I think not. Currently SSE is using the Bus to lookup data sources.

I think I need to update SSE to take a signed in user and data source cache to ensure it is always checked close to the query.

Reply:
good point! I think that the way datasource permissions are applied (given it's an enterprise feature) is that they override the datasource cache. In the similar endpoint for evaluating a condition I used to validate if all the queries referenced an existing datasource therefore I had [to query](https://github.com/grafana/grafana/blob/0dcd0281abbbb962697cca9ad30dd5ac43e1f859/pkg/services/ngalert/api/util.go#L185) the cache. I will add a similar check for this endpoint.

Reply:
Sounds good :+1: